source/simulator/assemblers/newton_stokes.cc

```diff
@@ -351,15 +358,21 @@ namespace aspect
                                         :
                                         1;
 
+                  // pre-compute the tensor contractions
+                  std::vector<SymmetricTensor<2,dim> > tensor_contraction(stokes_dofs_per_cell);
+                  for (unsigned int i = 0; i < stokes_dofs_per_cell; ++i)
+                    tensor_contraction[i] = (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[i]) * strain_rate;
+
+
                   // symmetrize when the stabilization is symmetric or SPD
                   if ((velocity_block_stabilization & Newton::Parameters::Stabilization::symmetric) != Newton::Parameters::Stabilization::none)
                     {
                       for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
                         for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
                           {
-                            data.local_matrix(i,j) += ( derivative_scaling_factor * alpha * (scratch.grads_phi_u[i] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[j]) * strain_rate
-                                                                                             + scratch.grads_phi_u[j] * (viscosity_derivative_wrt_strain_rate * scratch.grads_phi_u[i]) * strain_rate)
-                                                        + derivative_scaling_factor * this->get_pressure_scaling() * scratch.grads_phi_u[i] * 2.0 * viscosity_derivative_wrt_pressure * scratch.phi_p[j] * strain_rate )
+                            data.local_matrix(i,j) += ( derivative_scaling_factor * alpha * (scratch.grads_phi_u[i] * tensor_contraction[j]
+                                                                                             + scratch.grads_phi_u[j] * tensor_contraction[i])
+                                                        + derivative_scaling_factor * pressure_scaling * 2.0 * viscosity_derivative_wrt_pressure * scratch.phi_p[j] * scratch.grads_phi_u[i] * strain_rate )
```

Comment:
I can't quite convince myself that this change in the last line above is correct because you are reordering the multiplication of things I don't recall are all scalar. Either way, I think it would be nice to continue multiplying *first* with th i term and then with the j term as it was originally.